### ISSUE TITLE
Possible to generate this keyword?

### ISSUE THREAD

**Issue by lefthandedgoat** [fable-compiler/Fable#665]
### Description

How can I generate the javascript `this` keyword

### Repro code
I want to generate javascript like this

```
$('#the-element').on('click', { self : this }, function () { //whatever} );
```

The important part is that I can pass in an object with a self property, using the `this` keyword as it's value.

**Comment by alfonsogarciacaro** [comment]
Still not documented, sorry, but you can use `Fable.Core.JsInterop.jsThis`. At the moment it can only be used within a delegate but I may need to change this requirement because I think the pattern above is not possible at the moment (as JS `this` is called in an object).

Please also note that when you need to resort to that kind of JS tricks, it may make sense to write the JS code in a separate file and import it to your F# code: http://fable.io/docs/interacting.html#Import-attribute

**Comment by lefthandedgoat** [comment]
Ok thanks!

I am mediocre at best with js, and I am 'old school'.  It took me forever to get started with code that would run in the browser (needing to use webpack to bundle dependencies into a single file was the ultimate answer) but now that I have that part working, writing code with fable is pretty enjoyable.

Cheers, good job!

**Comment by alfonsogarciacaro** [comment]
Thanks! BTW, I published a new version (0.7.37) that allows you to use `jsThis` in module functions without the delegate. Also, Fable comes with Rollup embedded so if you don't need anything too complicated for the bundle you can just pass the `--rollup` flag to generate a single file (it'll will contain also the necessary code from fable-core). See: http://fable.io/blog/Introducing-0-7.html#ES2015-Modules-and-Bundling
